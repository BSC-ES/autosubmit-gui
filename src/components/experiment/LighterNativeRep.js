import React, { Component } from "react";
import Spinner from "../layout/Spinner";
import "jquery.fancytree/dist/modules/jquery.fancytree.filter";
import "jquery.fancytree/dist/modules/jquery.fancytree.childcounter";
import "jquery.fancytree/dist/modules/jquery.fancytree.edit";
import "jquery.fancytree/dist/modules/jquery.fancytree.multi";
// import 'jquery.fancytree/dist/skin-lion/ui.fancytree.less';  // CSS or LESS
import { createTree } from "jquery.fancytree";
import { DEBUG } from "../context/vars";

export class LighterNativeRep extends Component {
  shouldComponentUpdate(nextProps, nextState) {
    if (nextProps.data !== this.props.data) {
      DEBUG && console.log("Rerendering Lighter Tree");
      return true;
    } else if (nextProps.loadingView !== this.props.loadingView) {
      return true;
    } else {
      return false;
    }
  }

  componentWillUnmount() {
    DEBUG && console.log("Unmount Lighter Rep.");
    this.props.cleanLoadingLighterView();
  }

  componentDidMount() {
    DEBUG && console.log("Lighter Rep Mounted.");
  }

  render() {
    if (this.props.loadingView) return <Spinner></Spinner>;
    if (!this.props.data) {
      return (
        <div className='card-body text-left' style={experimentStyle}>
          <p className='lead'>
            Press <span className='badge badge-info'>View</span> to see the
            light-version of the experiment representation.
          </p>
          <p className='lead'>Use the search tool.</p>
        </div>
      );
    }
    if (this.props.data.error === true) {
      return (
        <div className='card-body text-left' style={experimentStyle}>
          <p className='lead'>Something has gone very wrong.</p>
          <p className='lead text-danger'>{this.props.data.error_message}</p>
        </div>
      );
    }

    class LightFancyTree extends Component {
      componentDidMount() {
        let l_tree = new createTree("#light_tree", {
          extensions: ["filter", "childcounter"],
          filter: {
            autoApply: true, // Re-apply last filter if lazy data is loaded
            autoExpand: true, // Expand all branches that contain matches while filtered
            counter: true, // Show a badge with number of matching child nodes near parent icons
            fuzzy: false, // Match single characters in order, e.g. 'fb' will match 'FooBar'
            hideExpandedCounter: true, // Hide counter badge if parent is expanded
            hideExpanders: false, // Hide expanders if all child nodes are hidden by filter
            highlight: false, // Highlight matches by wrapping inside <mark> tags
            leavesOnly: true, // Match end nodes only
            nodata: true, // Display a 'no data' status node if result is empty
            mode: "hide", // Grayout unmatched nodes (pass "hide" to remove unmatched node instead)
          },

          source: this.props.tree_data,
        });

        //console.log(tree.activeNode);

        DEBUG && console.log(l_tree);
        this.props.setLighterFancyTree(l_tree);
      }

      componentWillUnmount() {
        DEBUG && console.log("Unmounting Lighter Tree");
        //this.props.cleanNavData();
      }

      render() {
        return (
          <div className='card-body p-0'>
            <div id='light_tree'></div>
          </div>
        );
      }
    }

    return (
      <LightFancyTree
        tree_data={this.props.data.tree_view}
        originaldata={this.props.data}
        setLighterFancyTree={this.props.setLighterFancyTree}
      />
    );
  }
}

const experimentStyle = {
  height: 750,
};

export default LighterNativeRep;
