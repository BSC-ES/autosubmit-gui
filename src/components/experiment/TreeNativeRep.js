import React, { Component } from "react";
import Spinner from "../layout/Spinner";
import "jquery.fancytree/dist/modules/jquery.fancytree.clones";
import "jquery.fancytree/dist/modules/jquery.fancytree.filter";
import "jquery.fancytree/dist/modules/jquery.fancytree.childcounter";
import "jquery.fancytree/dist/modules/jquery.fancytree.edit";
import "jquery.fancytree/dist/modules/jquery.fancytree.multi";
// import 'jquery.fancytree/dist/skin-lion/ui.fancytree.less';  // CSS or LESS
import { createTree } from "jquery.fancytree";
import { DEBUG } from "../context/vars";

// import 'jquery.fancytree/dist/modules/jquery.fancytree.edit';
// import 'jquery.fancytree/dist/modules/jquery.fancytree.filter';

export class TreeNativeRep extends Component {
  shouldComponentUpdate(nextProps, nextState) {
    // if (this.props.shouldUpdateGraph === true){
    //     return true
    //   }else
    if (nextProps.treedata !== this.props.treedata) {
      DEBUG && console.log("Rerendering Tree");
      return true;
    } else if (nextProps.loadingTree !== this.props.loadingTree) {
      return true;
    } else {
      return false;
    }
  }

  componentWillUnmount() {
    DEBUG && console.log("Unmounting Tree Rep");
    this.props.cleanTreeData();
    // this.props.clearStats();
  }

  componenteDidMount() {
    DEBUG && console.log("In after mount");
    // createTree("#tree", {
    //   extensions: ["edit", "filter"],
    //   source: this.props.treedata,
    // });
  }

  render() {
    //console.log(this.props.treedata)
    if (this.props.loadingTree) return <Spinner></Spinner>;
    if (!this.props.treedata) {
      return (
        <div className='card-body text-left' style={experimentStyle}>
          <p className='lead'>
            Press <span className='badge badge-info'>Show Tree View</span> to
            see the tree view representation of the experiment.
          </p>
          <p className='lead'>
            Repeating subtrees in the tree view are only shown once, use the
            searcher to focus only on those items.
          </p>
        </div>
      );
    }

    if (this.props.treedata.error === true) {
      return (
        <div className='card-body text-left' style={experimentStyle}>
          <p className='lead'>Something has gone very wrong.</p>
          <p className='lead text-danger'>
            {this.props.treedata.error_message}
          </p>
        </div>
      );
    }

    class FancyTree extends Component {
      componentDidMount() {
        let tree = new createTree("#tree", {
          activate: (event, data) => {
            //console.log(event)
            //console.log(data)
            //console.log(tree)
            //console.log(data);
            if (data) {
              // var thenode = tree.getNodesByRef("a2a7_20170427_1")
              // console.log(thenode)
              // if (thenode){
              //     for (var i = 0; i < thenode.length;i++){
              //         thenode[i].setTitle("a2a7_20170417_1 <span class='badge badge-warning'> Honked </span>")
              //     }

              // }
              // console.log(tree.activeNode.getParent())
              // console.log(tree.activeNode)
              //data.node.title = "Honk";
              //console.log(data);
              //console.log(this);
              this.props.updateSelectionTree(data);
              //console.log(this.props.canSelect);
              //if (this.props.canSelect === true) {
              if (data && data.node && data.node.folder === undefined) {
                this.props.updateCurrentSelected(
                  data.node.refKey,
                  this.props.originaldata
                );
              }

              //}

              //this.updateSelection(data);
            }
          },
          // extensions: ['edit', 'filter'],
          // extensions: ["clones","filter", "childcounter"],
          extensions: ["filter", "childcounter", "clones"],
          filter: {
            autoApply: true, // Re-apply last filter if lazy data is loaded
            autoExpand: true, // Expand all branches that contain matches while filtered
            counter: true, // Show a badge with number of matching child nodes near parent icons
            fuzzy: false, // Match single characters in order, e.g. 'fb' will match 'FooBar'
            hideExpandedCounter: true, // Hide counter badge if parent is expanded
            hideExpanders: false, // Hide expanders if all child nodes are hidden by filter
            highlight: false, // Highlight matches by wrapping inside <mark> tags
            leavesOnly: true, // Match end nodes only
            nodata: true, // Display a 'no data' status node if result is empty
            mode: "hide", // Grayout unmatched nodes (pass "hide" to remove unmatched node instead)
          },
          // clones: {
          //     highlightClones: true,
          //     highlightActiveClones: true,
          //   },
          source: this.props.treedata,
        });

        //console.log(tree.activeNode);

        DEBUG && console.log(tree);
        this.props.setFancyTree(tree);
      }

      componentWillUnmount() {
        DEBUG && console.log("Unmounting Tree");
        //this.props.cleanNavData();
      }

      render() {
        return (
          <div className='card-body p-0'>
            <div id='tree'></div>
          </div>
        );
      }
    }

    return (
      <FancyTree
        treedata={this.props.treedata.tree}
        originaldata={this.props.treedata}
        setFancyTree={this.props.setFancyTree}
        updateSelectionTree={this.props.updateSelectionTree}
        updateCurrentSelected={this.props.updateCurrentSelected}
        canSelect={this.props.canSelect}
      />
    );
  }
}

// const experimentStyle = {
//     maxHeight: 600,
//     height: 600
//   };

const experimentStyle = {
  height: 750,
};

export default TreeNativeRep;
