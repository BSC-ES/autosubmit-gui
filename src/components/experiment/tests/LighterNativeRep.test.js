import React from "react";
import { unmountComponentAtNode, render } from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import TreeNativeRep from "../TreeNativeRep";
import ExperimentContext from "../../context/experiment/experimentContext";
import LighterNativeRep from "../LighterNativeRep";

let jobs_included = [];
const lightData = {
  "completed": 18,
  "error": false,
  "error_message": "",
  "failed": 0,
  "queuing": 0,
  "running": 1,
  "total": 19,
  "tree_view": [
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_LOCAL_SETUP",
          "title": "t0ie_LOCAL_SETUP <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_REMOTE_COMPILE",
          "title": "t0ie_REMOTE_COMPILE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_PREPROCVAR",
          "title": "t0ie_20120101_000_2_PREPROCVAR <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_REDUCE",
          "title": "t0ie_20120101_000_2_REDUCE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_ARCHIVE",
          "title": "t0ie_20120101_000_2_ARCHIVE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_LOCAL_SEND_INITIAL",
          "title": "t0ie_20120101_000_2_LOCAL_SEND_INITIAL <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_ARCHIVE",
          "title": "t0ie_20120101_000_1_ARCHIVE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_PREPROCVAR",
          "title": "t0ie_20120101_000_1_PREPROCVAR <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_LOCAL_SEND_SOURCE",
          "title": "t0ie_LOCAL_SEND_SOURCE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_CLEAN",
          "title": "t0ie_20120101_000_1_CLEAN <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_2_ARCHIVE_REDUCE",
          "title": "t0ie_20120101_2_ARCHIVE_REDUCE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_LOCAL_SEND_STATIC",
          "title": "t0ie_LOCAL_SEND_STATIC <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_LOCAL_SEND_INITIAL",
          "title": "t0ie_20120101_000_1_LOCAL_SEND_INITIAL <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_1_ARCHIVE_REDUCE",
          "title": "t0ie_20120101_1_ARCHIVE_REDUCE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_CLEAN",
          "title": "t0ie_20120101_000_2_CLEAN <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_PREPROCFIX",
          "title": "t0ie_PREPROCFIX <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_2_SIM",
          "title": "t0ie_20120101_000_2_SIM <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_SIM",
          "title": "t0ie_20120101_000_1_SIM <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      },
      {
          "children": [],
          "data": "Empty",
          "refKey": "t0ie_20120101_000_1_REDUCE",
          "title": "t0ie_20120101_000_1_REDUCE <span class='badge' style='background-color: yellow'>#COMPLETED</span>"
      }
  ],
  "view_data": [
      {
          "err": "/t0ie_LOCAL_SETUP.20201109110038.err",
          "name": "t0ie_LOCAL_SETUP",
          "out": "/t0ie_LOCAL_SETUP.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_REMOTE_COMPILE.20201109110038.err",
          "name": "t0ie_REMOTE_COMPILE",
          "out": "/t0ie_REMOTE_COMPILE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_PREPROCVAR.20201109110038.err",
          "name": "t0ie_20120101_000_2_PREPROCVAR",
          "out": "/t0ie_20120101_000_2_PREPROCVAR.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_REDUCE.20201109110038.err",
          "name": "t0ie_20120101_000_2_REDUCE",
          "out": "/t0ie_20120101_000_2_REDUCE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_ARCHIVE.20201109110038.err",
          "name": "t0ie_20120101_000_2_ARCHIVE",
          "out": "/t0ie_20120101_000_2_ARCHIVE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_LOCAL_SEND_INITIAL.20201109110038.err",
          "name": "t0ie_20120101_000_2_LOCAL_SEND_INITIAL",
          "out": "/t0ie_20120101_000_2_LOCAL_SEND_INITIAL.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_ARCHIVE.20201109110038.err",
          "name": "t0ie_20120101_000_1_ARCHIVE",
          "out": "/t0ie_20120101_000_1_ARCHIVE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_PREPROCVAR.20201109110038.err",
          "name": "t0ie_20120101_000_1_PREPROCVAR",
          "out": "/t0ie_20120101_000_1_PREPROCVAR.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_LOCAL_SEND_SOURCE.20201109110038.err",
          "name": "t0ie_LOCAL_SEND_SOURCE",
          "out": "/t0ie_LOCAL_SEND_SOURCE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_CLEAN.20201109110038.err",
          "name": "t0ie_20120101_000_1_CLEAN",
          "out": "/t0ie_20120101_000_1_CLEAN.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_2_ARCHIVE_REDUCE.20201109110038.err",
          "name": "t0ie_20120101_2_ARCHIVE_REDUCE",
          "out": "/t0ie_20120101_2_ARCHIVE_REDUCE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_LOCAL_SEND_STATIC.20201109110038.err",
          "name": "t0ie_LOCAL_SEND_STATIC",
          "out": "/t0ie_LOCAL_SEND_STATIC.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_LOCAL_SEND_INITIAL.20201109110038.err",
          "name": "t0ie_20120101_000_1_LOCAL_SEND_INITIAL",
          "out": "/t0ie_20120101_000_1_LOCAL_SEND_INITIAL.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_1_ARCHIVE_REDUCE.20201109110038.err",
          "name": "t0ie_20120101_1_ARCHIVE_REDUCE",
          "out": "/t0ie_20120101_1_ARCHIVE_REDUCE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_CLEAN.20201109110038.err",
          "name": "t0ie_20120101_000_2_CLEAN",
          "out": "/t0ie_20120101_000_2_CLEAN.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_PREPROCFIX.20201109110038.err",
          "name": "t0ie_PREPROCFIX",
          "out": "/t0ie_PREPROCFIX.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_2_SIM.20201109110038.err",
          "name": "t0ie_20120101_000_2_SIM",
          "out": "/t0ie_20120101_000_2_SIM.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_SIM.20201109110038.err",
          "name": "t0ie_20120101_000_1_SIM",
          "out": "/t0ie_20120101_000_1_SIM.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      },
      {
          "err": "/t0ie_20120101_000_1_REDUCE.20201109110038.err",
          "name": "t0ie_20120101_000_1_REDUCE",
          "out": "/t0ie_20120101_000_1_REDUCE.20201109110038.out",
          "path_log": "/esarchive/autosubmit/t0ie/tmp/LOG_t0ie"
      }
  ]
};

lightData.tree_view.map(item => jobs_included.push(item.refKey));

const experiment = {expid: "t0ie", db_historic_version: 14, owner: "wuruchi" };
let treeRep = null;

let container = null;
beforeEach(() => {
  // setup DOM element as render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("Lighter Representation renders with content", () => {
   
  act(() => {
    render(<LighterNativeRep data={lightData} loadingView={false} setLighterFancyTree={(obj) => treeRep = obj} cleanLoadingLighterView={() => null} />, container);
  });
  
  // console.log(treeRep);
  // console.log(container.innerHTML);
  // console.log(container.textContent)
  jobs_included.forEach(item => expect(container.textContent).toContain(item));

});

it("Lighter Representation matches snapshot", () => {


  const tree = renderer.create(<ExperimentContext.Provider value={{expectedLoadingTreeTime:() => 10}}><LighterNativeRep data={lightData} loadingView={false} setLighterFancyTree={(obj) => treeRep = obj} cleanLoadingLighterView={() => null} /></ExperimentContext.Provider> ).toJSON();

  expect(tree).toMatchSnapshot();

});

