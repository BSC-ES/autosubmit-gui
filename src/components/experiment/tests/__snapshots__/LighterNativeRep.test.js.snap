// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lighter Representation matches snapshot 1`] = `
<div
  className="card-body p-0"
>
  <div
    id="light_tree"
  />
</div>
`;
